src/index.rs

```diff
@@ -83,29 +82,62 @@ impl ResourceIndex {
         for line in BufReader::new(file).lines() {
             if let Ok(entry) = line {
                 let mut parts = entry.split(' ');
 
                 let modified: SystemTime = {
-                    let str = parts.next().ok_or(ArklibError::Parse)?;
-                    UNIX_EPOCH.add(Duration::from_millis(
-                        str.parse().map_err(|_| ArklibError::Parse)?,
-                    ))
+                    match parts.next() {
+                        Some(millis) => match millis.parse() {
+                            Ok(millis) => {
+                                UNIX_EPOCH.add(Duration::from_millis(millis))
+                            }
+                            Err(_) => {
+                                log::warn!("Can't parse {millis} to u64");
+                                continue;
+                            }
+                        },
+                        None => {
+                            log::warn!("Missing timestamp from {entry}");
+                            continue;
+                        }
+                    }
                 };
 
                 let id: ResourceId = {
-                    let str = parts.next().ok_or(ArklibError::Parse)?;
-                    ResourceId::from_str(str)?
+                    match parts.next() {
+                        Some(str) => match ResourceId::from_str(str) {
+                            Ok(id) => id,
+                            Err(_) => {
+                                log::warn!(
+                                    "Can't parse {str} to a RessourceId"
+                                );
+                                continue;
+                            }
+                        },
+                        None => {
+                            log::warn!(
+                                "Missing ressource id part on the line {entry}"
+                            );
+                            continue;
+                        }
+                    }
                 };
 
-                let path: String = parts.intersperse(" ").collect();
+                let path: String =
+                    itertools::Itertools::intersperse(parts, " ").collect();
                 let path: PathBuf = root_path.join(Path::new(&path));
-                let path: CanonicalPathBuf =
-                    CanonicalPathBuf::canonicalize(path)?;
-
-                log::trace!("[load] {} -> {}", id, path.display());
-                index.insert_entry(path, IndexEntry { id, modified });
+                match CanonicalPathBuf::canonicalize(&path) {
+                    Ok(path) => {
+                        log::trace!("[load] {} -> {}", id, path.display());
+                        index.insert_entry(path, IndexEntry { id, modified });
+                    }
+                    Err(_) => {
+                        log::warn!("File {} not found", path.display());
+                        continue;
+                    }
```

Comment:
Here continuing should be good :+1:

Reply:
Ok I've updated to return early if the file can't be parsed